database/manager.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
+)
+
+//
+// DatabaseManager writes all committed keys to the database on a timer.
+//
+
+type DatabaseManager struct {
+	logger      logging.Logger
+	keyManagers map[RelayerID]*keyManager
+	interval    time.Duration
+	db          RelayerDatabase
+}
+
+func NewDatabaseManager(logger logging.Logger, db RelayerDatabase, interval time.Duration, keys []RelayerID) *DatabaseManager {
+	keyManagers := make(map[RelayerID]*keyManager)
+	for _, key := range keys {
+		keyManager := newKeyManager(logger)
+		go keyManager.run()
+		keyManagers[key] = keyManager
+	}
+	return &DatabaseManager{
+		logger:      logger,
+		db:          db,
+		interval:    interval,
+		keyManagers: keyManagers,
+	}
+}
+
+// Run writes all committed keys to the database on the configured timer.
+// This function should only be called once.
+func (dm *DatabaseManager) Run() {
+	for range time.Tick(dm.interval) {
+		for id, km := range dm.keyManagers {
```

Comment:
actually doing a quick readup seems like the ticks continue in background, and if the underlying for loop is not completed, then the tick iteration will get dropped and waits for the next. Is that a problem at all for short durations?

Reply:
The tick interval is on the order of seconds (or 10's of seconds), so the only expected scenario this could occur is if there's network issues reaching the remote database. Even so, this isn't really a problem since a missed write to the database is perfectly safe.